sandbox/index.js

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable flowtype/require-valid-file-annotation */
+const fs = require('fs')
+
+const path = require('path')
+
+const express = require('express')
+
+const webpackDevMiddleware = require('webpack-dev-middleware')
+
+const webpack = require('webpack')
+
+const { CONFIG, logger } = require('./util')
+
+const webpackConfig = require('./webpack.config.dev')
+
+const livereloadMiddleware = require('./livereload/middleware')
+
+const compiler = webpack(webpackConfig)
+
+const devMiddleware = webpackDevMiddleware(compiler, webpackConfig.devServer)
+
+const indexHtml = fs.readFileSync(
+  path.resolve(__dirname, './public/index.html'),
+  'utf-8' // eslint-disable-line comma-dangle
+)
+
+const app = express()
+
+const server = require('http').Server(app)
+
+app.use(devMiddleware)
+
+app.use(livereloadMiddleware(compiler, server))
+
+app.get('*', (req, res) => {
+  // We use webpackDevMiddleware in-memory file system to read server script after compilation
+  const serverScript = devMiddleware.fileSystem.readFileSync(
+    path.join(webpackConfig.output.path, 'server.js'),
+    'utf-8' // eslint-disable-line comma-dangle
+  )
+
+  // Then we evaluate this script to get prerendered content
+  const { html, css } = eval(serverScript) // eslint-disable-line no-eval
```

Comment:
I have some setup code lying around in one of my repos (which works like next.js) that uses the built-in hot reloader, dev middleware, and compiles both the server and the client in tandem without evaling.

I can probably implement this in a future separate PR, because I'm not comfortable having this approach here permanently, as some people might copy it for their own setups

Reply:
If you want, I'm happy to help with this in a future PR, if you have some examples to look at 😉